### PR TITLE
[plugin.video.drnu] v5.0.13

### PR DESCRIPTION
**Pull request by TermeHansen**
### Description
Update (only tested on Leia, 5.0.12 exists in Gotham branch)
- Adding old PR's + minor fixes. last update before kodi 19 and python3
- Change for request-cache for storing web requests locally (in 8hours)
- Fix missing thumbnails and added option in settings to choose fan-art image size
- Fixed themes menu item
- Add support for other plugins to start live tv.
- Update logos and entry screen
- Change subtitle handling to local srt files, so user also can change for hard-of-hearing subs and change while watching

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.drnu/addon.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon id="plugin.video.drnu" version="5.0.13" name="DR TV" provider-name="TermeHansen">
+    <requires>
+        <import addon="xbmc.python" version="2.26.0"/>
+        <import addon="script.module.requests" version="2.22.0" />
+        <import addon="script.module.requests-cache" version="0.4.13"/>
+        <import addon="script.module.simplejson" version="2.0.10"/>
```
Comment (enen92) [inline plugin.video.drnu/addon.xml:7]: please remove the simplejson dependency it is not needed anymore (this was only needed for helix)

File: plugin.video.drnu/addon.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en_GB">Watch live and archived TV from DR (Danish Broadcasting Corporation)</summary>
+        <summary lang="da_DK">Se live og arkiveret TV fra DR</summary>
+        <description lang="en_GB">All DR's live TV channels and archived programs as seen on dr.dk/tv.[CR][CR]If you have comments or suggestions for this addon, please feel free to participate in the KODI official forums</description>
+        <description lang="da_DK">Alle DR's live TV kanaler samt alle de programmer du finder på dr.dk/tv.[CR][CR]Har du kommentarer, ris eller ros til denne addon er du velkommen til at deltage i debatten på KODI forummet</description>
+        <disclaimer lang="en_GB">Some parts of this addon may not be legal in your country of residence - please check with your local laws.</disclaimer>
+        <disclaimer lang="da_DK">Nogle dele af denne addon er muligvis ikke lovlig i dit land - kontroller venligst dine lokale love.</disclaimer>
+        <license>GPL 2.0</license>
```
Comment (enen92) [inline plugin.video.drnu/addon.xml:20]: "GPL-2.0-only" (spdx identifier)

File: plugin.video.drnu/addon.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        <platform>all</platform>
+        <language>da</language>
+        <email>termopeten@gmail.com</email>
+        <website>https://www.dr.dk/</website>
+        <forum>https://forum.kodi.tv/</forum>
```
Comment (enen92) [inline plugin.video.drnu/addon.xml:25]: you should either open a forum post for support and link to ir or drop the forum element altogether

File: plugin.video.drnu/addon.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        <source>https://github.com/xbmc-danish-addons/plugin.video.drnu</source>
```
Comment (enen92) [inline plugin.video.drnu/addon.xml:26]: This is the upstream of the code. Afaics the newer version of the addon is not here anymore. Please adjust and point to your repo.

File: plugin.video.drnu/resources/lib/addon.py [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+
+import buggalo
+
+tr = xbmcaddon.Addon().getLocalizedString
+get_setting = xbmcaddon.Addon().getSetting
+addon_path = xbmcaddon.Addon().getAddonInfo('path')
+addon_name = xbmcaddon.Addon().getAddonInfo('name')
+
+def make_notice(object):
+    xbmc.log(str(object), xbmc.LOGNOTICE )
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:44]: This needs to be LOGDEBUG by default

File: plugin.video.drnu/resources/lib/addon.py [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+            self.showMainMenu()
+        else:
+            items = self.api.getChildrenFrontItems('dr-' + areaSelected)
+            #xbmc.executebuiltin('Container.SetViewMode(500)')
+            self.listSeries(items)
+
+    def showMainMenu(self):
+        items = list()
+        # Live TV
+        item = xbmcgui.ListItem(tr(30027), iconImage=os.path.join(addon_path, 'resources', 'icons', 'livetv.png'))
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:111]: please remove the iconimage and the setproperty(fanart image) both are deprecated.
Use:

`item.setArt({"icon":"xxx", "fanart": "xxx"})`

File: plugin.video.drnu/resources/lib/addon.py [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+        item.setProperty('Fanart_Image', self.fanart_image)
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=liveTV', item, True))
+
+        # A-Z Program Series
+        item = xbmcgui.ListItem(tr(30000), iconImage=os.path.join(addon_path, 'resources', 'icons', 'all.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:118]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=listAZ', item, True))
+
+        # Latest
+        item = xbmcgui.ListItem(tr(30001), iconImage=os.path.join(addon_path, 'resources', 'icons', 'all.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:124]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 9]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=latest', item, True))
+
+        # Premiere
+        item = xbmcgui.ListItem(tr(30025), iconImage=os.path.join(addon_path, 'resources', 'icons', 'new.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:130]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 10]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?listVideos=%s' % tvapi.SLUG_PREMIERES, item, True))
+
+        # Themes
+        item = xbmcgui.ListItem(tr(30028), iconImage=os.path.join(addon_path, 'resources', 'icons', 'all.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:136]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 11]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=themes', item, True))
+
+        # Most viewed
+        item = xbmcgui.ListItem(tr(30011), iconImage=os.path.join(addon_path, 'resources', 'icons', 'eye.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:142]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 12]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=mostViewed', item, True))
+
+        # Spotlight
+        item = xbmcgui.ListItem(tr(30002), iconImage=os.path.join(addon_path, 'resources', 'icons', 'star.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:148]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 13]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=highlights', item, True))
+
+        # Search videos
+        item = xbmcgui.ListItem(tr(30003), iconImage=os.path.join(addon_path, 'resources', 'icons', 'search.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:154]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 14]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=search', item, True))
+
+        # Recently watched Program Series
+        item = xbmcgui.ListItem(tr(30007), iconImage=os.path.join(addon_path, 'resources', 'icons', 'eye-star.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:160]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 15]
```diff
@@ -0,0 +1,508 @@
...
+        item.addContextMenuItems(self.menuItems, False)
+        items.append((self._plugin_url + '?show=recentlyWatched', item, True))
+
+        # Favorite Program Series
+        item = xbmcgui.ListItem(tr(30008), iconImage=os.path.join(addon_path, 'resources', 'icons', 'plusone.png'))
+        item.setProperty('Fanart_Image', self.fanart_image)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:166]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 16]
```diff
@@ -0,0 +1,508 @@
...
+
+        xbmcplugin.addDirectoryItems(self._plugin_handle, items)
+        xbmcplugin.endOfDirectory(self._plugin_handle)
+
+    def showAZ(self):
+        # All Program Series
+        iconImage = os.path.join(addon_path, 'resources', 'icons', 'all.png')
+        items = list()
+        for programIndex in self.api.getProgramIndexes():
+            item = xbmcgui.ListItem(programIndex['Title'], iconImage=iconImage)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:241]: drop iconimage arg, use setArt

File: plugin.video.drnu/resources/lib/addon.py [hunk 17]
```diff
@@ -0,0 +1,508 @@
...
+            items.append((url, item, True))
+        xbmcplugin.addDirectoryItems(self._plugin_handle, items)
+        xbmcplugin.endOfDirectory(self._plugin_handle)
+
+    def showThemes(self):
+        iconImage = os.path.join(addon_path, 'resources', 'icons', 'all.png')
+
+        items = list()
+        for theme in self.api.getThemes():
+            item = xbmcgui.ListItem(theme['Title'], iconImage=iconImage)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:255]: same for iconimage and fanart, use setArt

File: plugin.video.drnu/resources/lib/addon.py [hunk 18]
```diff
@@ -0,0 +1,508 @@
...
+                                    tr(30018), tr(30019))
+            else:
+                xbmcgui.Dialog().ok(addon_name, tr(30013))
+        else:
+            directoryItems = list()
+            for item in items:
+                menuItems = list(self.menuItems)
+
+                if self.favorites.count(item['SeriesTitle']) > 0:
+                    runScript = "XBMC.RunPlugin(plugin://plugin.video.drnu/?delfavorite=%s)" % item['SeriesTitle'].replace('&', '%26').replace(',', '%2C')
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:286]: drop xbmc. prefix, use only "RunPlugin("

File: plugin.video.drnu/resources/lib/addon.py [hunk 19]
```diff
@@ -0,0 +1,508 @@
...
+                    menuItems.append((tr(30201), runScript))
+                else:
+                    runScript = "XBMC.RunPlugin(plugin://plugin.video.drnu/?addfavorite=%s)" % item['SeriesTitle'].replace('&', '%26').replace(',', '%2C')
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:289]: same

File: plugin.video.drnu/resources/lib/addon.py [hunk 20]
```diff
@@ -0,0 +1,508 @@
...
+                server = None
+                for streamingServer in channel['StreamingServers']:
+                    if streamingServer['LinkType'] == 'HLS':
+                        server = streamingServer
+                        break
+                if server is None:
+                    continue
+
+                url = server['Server'] + '/' + server['Qualities'][0]['Streams'][0]['Stream']
+                item = xbmcgui.ListItem(channel['Title'], iconImage=channel['PrimaryImageUri'], path=url)
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/addon.py:378]: same, setArt

File: plugin.video.drnu/resources/lib/tvapi.py [hunk 21]
```diff
@@ -0,0 +1,490 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with XBMC; see the file COPYING.  If not, write to
+#  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+#  http://www.gnu.org/copyleft/gpl.html
+#
+
+try:
```
Comment (enen92) [inline plugin.video.drnu/resources/lib/tvapi.py:22]: use json only, drop simplejson

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments. Loads of deprecated/outdated stuff.
PS: Is the AES decrypt/encrypt really required?

File: plugin.video.drnu/addon.xml (hunk 4 above)
Comment (TermeHansen) [reply plugin.video.drnu/addon.xml:26]: nope, this where I and others have taken over after Tommy decided to leave the project...

**Pull request closed without merging**